src/parser/serializer.ts

```diff
@@ -368,7 +369,7 @@ function serializeBuffer(
   index = index + numberOfWrittenBytes;
   buffer[index++] = 0;
   // Get size of the buffer (current write point)
-  const size = value.length;
+  const size = value.byteLength;
```

Comment:
was this just a bug? There is a difference between these two values

Reply:
Now that any TypedArrays land in this path of serialization it's important to use the byteLength. `Float32Array([1.2, 3.4])` has length 2 but byteLength 8. If it were only int8 types arriving here then length and  byteLength would be the same.

Reply:
> Now that any TypedArrays land in this path

Just in terms of typing right? TypedArrays have always been able to take this path. Does this mean we've been serializing them incorrectly all along? Might be a case for making more of a breaking change if it never worked correctly in the first place

Reply:
There was a modification in the serialization that led to this (because the type annotations followed the change it was having me change this to byteLength) this has be fixed